be/src/olap/comparison_predicate.cpp

```diff
@@ -145,28 +146,65 @@ COMPARISON_PRED_COLUMN_BLOCK_EVALUATE(LessEqualPredicate, <=)
+                    for (uint16_t i = 0; i < *size; i++) {                                         \
+                        uint16_t idx = sel[i];                                                     \
+                        sel[new_size] = idx;                                                       \
+                        const auto& cell_value =                                                   \
+                                reinterpret_cast<const vectorized::Int32&>(data_array[idx]);       \
+                        bool ret = !null_bitmap[idx] && (cell_value OP code);                      \
+                        new_size += _opposite ? !ret : ret;                                        \
+                    }                                                                              \
+                }                                                                                  \
+            } else {                                                                               \
+                auto* nested_col_ptr =                                                             \
+                        vectorized::check_and_get_column<vectorized::PredicateColumnType<type>>(   \
+                                nested_col);                                                       \
+                auto& data_array = nested_col_ptr->get_data();                                     \
+                for (uint16_t i = 0; i < *size; i++) {                                             \
+                    uint16_t idx = sel[i];                                                         \
+                    sel[new_size] = idx;                                                           \
+                    const type& cell_value = reinterpret_cast<const type&>(data_array[idx]);       \
+                    bool ret = !null_bitmap[idx] && (cell_value OP _value);                        \
+                    new_size += _opposite ? !ret : ret;                                            \
+                }                                                                                  \
             }                                                                                      \
             *size = new_size;                                                                      \
+        } else if (column.is_column_dictionary()) {                                                \
+            if constexpr (std::is_same_v<type, StringValue>) {                                     \
+                auto& dict_col =                                                                   \
+                        reinterpret_cast<vectorized::ColumnDictionary<vectorized::Int32>&>(column);\
+                auto& data_array = dict_col.get_data();                                            \
+                auto code = IS_RANGE ? dict_col.find_bound_code(_value, 0 OP 1, 1 OP 1)            \
```

Comment:
why we do not dispose value before really do predicate, not need hash string value again and again, to speed up query

Reply:
Great idea, I'm about to improve it too, by rewriting the predicate so that it runs vectorized predicate computations.

Reply:
Great idea, I'm about to improve it too, by rewriting the predicate so that it runs vectorized predicate computations.